pre-commit: handle *.md files with mdformat + apply it

dbast opened the pull request:
Also exclude .md files from trailing-whitespace hook and update hooks via `pre-commit autoupdate`.

closes https://github.com/michalfita/packer-plugin-cross/issues/7

michalfita commented on README.md:
```diff
@@ -61,52 +57,64 @@ This method is primarily for macOS users where is no native way to use `qemu-use
 The container is a multi-arch container (linux/amd64 or linux/arm64), that can be used on Intel (x86_64) or Apple M1 (arm64) Macs and also on Linux machines running linux (x86_64 or aarch64) kernels.
 
-> ![NOTE]
+> !\[NOTE\]
```
Looks like `gfm` plugin doesn't understand notice boxes supported by GitHub for about a year now.

dbast replied:
right... didn't see that... also seems to be not part of the spec https://github.github.com/gfm/ ... maybe prettier as hook is then more generic and formats .md/, .json, .yaml.

michalfita replied:
Raised hukkin/mdformat-gfm#29 about missing support for this nice looking and useful feature.

dbast replied:
thanks!

michalfita reviewed (approved):
I have doubts about details of results, but in principle this is correct approach.

dbast committed:
pre-commit: handle *.md files with mdformat + apply it

Also exclude .md files from trailing-whitespace hook and update hooks via
`pre-commit autoupdate`.

dbast committed:
Fix !NOTE syntax

dbast commented:
@michalfita This is now ready again, as the `!\[NOTE\]` syntax issue could be fixed by using `\[!NOTE\]`, which avoids the mdformat issue. I guess https://github.com/hukkin/mdformat-gfm/issues/29 can be also closed.
